src/TextareaAutosize.js

```diff
@@ -87,6 +93,13 @@ export default class TextareaAutosize extends React.Component {
     // Re-render with the new content then recalculate the height as required.
     this.onNextFrameActionId = onNextFrame(this._resizeComponent);
   }
+  
+  componentWillUpdate(nextProps, nextState) {
```

Comment:
I agree with you calling `setState` from within componentWillUpdate will cause you troubles. 
Modifying Textarea's height from parent's callback will do the same. It's up to developer to use the callback properly, right?